### PR TITLE
roon-server: revamp

### PR DESCRIPTION
**Pull request by lovesegfault**
###### Motivation for this change
This builds on https://github.com/NixOS/nixpkgs/pull/92945 by completely revamping the `roon-server` derivation, and in the process fixing numerous issues with it.

With these changes I now have a fully functioning Roon on NixOS setup for the first time!

cc. @eljojo
cc. @flokli @andir 

###### Things done

- [x] Tested using sandboxing ([nix.useSandbox](https://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](https://nixos.org/nix/manual/#sec-conf-file) on non-NixOS linux)
- Built on platform(s)
   - [x] NixOS
   - [ ] macOS
   - [ ] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [ ] Tested compilation of all pkgs that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review wip"`
- [x] Tested execution of all binary files (usually in `./result/bin/`)
- [ ] Determined the impact on package closure size (by running `nix path-info -S` before and after)
- [x] Ensured that relevant documentation is up to date
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

### CODE REVIEW

File: pkgs/servers/roon-server/default.nix [hunk 1]
```diff
@@ -1,65 +1,68 @@
...
-    sed -i '/ln -sf/ d' Server/RoonServer
-    mkdir -p $out/opt
-    mv * $out/opt
-    ln -sf $out/opt/RoonMono/bin/mono-sgen $out/opt/RoonMono/bin/RoonServer
-    ln -sf $out/opt/RoonMono/bin/mono-sgen $out/opt/RoonMono/bin/RoonAppliance
-    ln -sf $out/opt/RoonMono/bin/mono-sgen $out/opt/RoonMono/bin/RAATServer
+  nativeBuildInputs = [ autoPatchelfHook makeWrapper ];
 
+  installPhase = ''
+    runHook preInstall
```
Comment (Ekleog) [inline pkgs/servers/roon-server/default.nix:33]: I'd bet the `runHook`s aren't actually required for proper operation... are they?

**Review by Ekleog — APPROVED** [review]

File: pkgs/servers/roon-server/default.nix (hunk 1 above)
Comment (lovesegfault) [reply pkgs/servers/roon-server/default.nix:33]: Probably not, but because I'm overriding the whole phase I thought I'd keep them.

**Comment by vcunat** [comment]
The zlib hack from #92945 isn't required anymore, right?

**Comment by Ekleog** [comment]
As far as I understand this is taken care of by the patchelf hook — that said I must say I am not a user of roon myself, so couldn't confirm with 100% certainty.

**Comment by eljojo** [comment]
Hi everyone, I'm glad that the roon-server package is getting more attention.

I have to admit it makes me feel a bit sad that this PR was merged before mine, I worked really hard on it and was hoping it'd get merged. 

With that being said, would someone be able to explain to me:
- how this goes about solving the zlib issue?
  - I went through the diff but still can't understand how this fixes the issue
  - I am not doubting that this does indeed fix it, I just want to improve my understanding of nix
- any guidance on how to run this on my local NixOS?
  - it seems superior than what I already have running, but I am not sure how to point my local NixOS to the "master branch" version of this roon-server package.

Thanks for your help and doing this work

**Comment by vcunat** [comment]
> any guidance on how to run this on my local NixOS?

There are various ways, e.g. use parameter `-I nixpkgs=/path` (accepted by `nixos-rebuild`, `nix-*` and even new `nix`), where `/path` can even be like https://github.com/nixos/nixpkgs/archive/master.tar.gz

**Comment by lovesegfault** [comment]
@vcunat It's being handled by the autoPatchelfHook.

@eljojo 
> how this goes about solving the zlib issue? 

The original Nix code was a bit "traditional" in that it manually tried to patchelf binaries, and so missed patchelf-ing zlib into the mono runtime. My changes uses the `autoPatchelfHook` as opposed to manually calling `patchelf`. What this effectively does is scan all ELF binaries for libraries and automatically patchelf the correct `rpath`s in. Thus not only the `zlib` issue you had is fixed, but also the `alsalib` issue I had, and any other new dynlinked dependencies Roon might add in the future.

> any guidance on how to run this on my local NixOS? 

You can add it in an overlay, but i think you'd benefit from going to `#nixos` on freenode for more interactive help on how to set those things up

**Comment by eljojo** [comment]
👍 thanks for the context this is super useful!

**Pull request merged**